presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/TestSpatialJoinOperator.java

```diff
@@ -182,8 +194,74 @@ public void testEmptyProbe()
         assertOperatorEquals(joinOperatorFactory, driverContext, probePages.build(), expected);
     }
 
-    private PagesSpatialIndexFactory buildIndex(DriverContext driverContext, BiPredicate<OGCGeometry, OGCGeometry> spatialRelationshipTest, RowPagesBuilder buildPages)
+    @Test
+    public void testYield()
     {
+        // create a filter function that yields for every probe match
+        // verify we will yield #match times totally
+
+        TaskContext taskContext = createTaskContext();
+        DriverContext driverContext = taskContext.addPipelineContext(0, true, true).addDriverContext();
+
+        // force a yield for every match
+        AtomicInteger filterFunctionCalls = new AtomicInteger();
+        InternalJoinFilterFunction filterFunction = new TestInternalJoinFilterFunction((
+                (leftPosition, leftBlocks, rightPosition, rightBlocks) -> {
+                    filterFunctionCalls.incrementAndGet();
+                    driverContext.getYieldSignal().forceYieldForTesting();
+                    return true;
+                }));
+
+        RowPagesBuilder buildPages = rowPagesBuilder(ImmutableList.of(GEOMETRY, VARCHAR))
+                .row(POLYGON_A, "A")
+                .row(POLYGON_B, "B");
+        PagesSpatialIndexFactory pagesSpatialIndexFactory = buildIndex(driverContext, (build, probe) -> build.contains(probe), Optional.of(filterFunction), buildPages);
+
+        // 10 points in polygon A (x0...x9)
+        // 10 points in polygons A and B (y0...y9)
+        // 10 points in polygon B (z0...z9)
+        // 40 total matches
+        RowPagesBuilder probePages = rowPagesBuilder(ImmutableList.of(GEOMETRY, VARCHAR));
+        for (int i = 0; i < 10; i++) {
+            probePages.row(stPoint(1 + 0.1 * i, 1 + 0.1 * i), "x" + i);
+        }
+        for (int i = 0; i < 10; i++) {
+            probePages.row(stPoint(4.5 + 0.01 * i, 4.5 + 0.01 * i), "y" + i);
+        }
+        for (int i = 0; i < 10; i++) {
+            probePages.row(stPoint(6 + 0.1 * i, 6 + 0.1 * i), "z" + i);
+        }
+        List<Page> probeInput = probePages.build();
+
+        OperatorFactory joinOperatorFactory = new SpatialJoinOperatorFactory(2, new PlanNodeId("test"), probePages.getTypes(), Ints.asList(1), 0, pagesSpatialIndexFactory);
+
+        Operator operator = joinOperatorFactory.createOperator(driverContext);
+        assertTrue(operator.needsInput());
+        operator.addInput(probeInput.get(0));
+        operator.finish();
+
+        // we will yield 40 times due to filterFunction
+        for (int i = 0; i < 40; i++) {
+            driverContext.getYieldSignal().setWithDelay(5 * SECONDS.toNanos(1), driverContext.getYieldExecutor());
+            filterFunctionCalls.set(0);
+            assertNull(operator.getOutput());
+            assertEquals(filterFunctionCalls.get(), 1, "Expected join to stop processing (yield) after calling filter function once");
```

Comment:
I found @findepi modified the logic in the test for hash join. Can't we just assert the call count equals to `i` and remove the `set(0)` part?